pkg/resource/certificate_authority_activation/sdk.go

```diff
@@ -114,6 +114,24 @@ func (rm *resourceManager) sdkCreate(
 			return nil, err
 		}
 	}
+
+	if desired.ko.Spec.Status != nil && *desired.ko.Spec.Status == svcsdk.CertificateAuthorityStatusDisabled {
+		updateInput := &svcsdk.UpdateCertificateAuthorityInput{}
+
+		updateInput.SetStatus(*desired.ko.Spec.Status)
+
+		if desired.ko.Spec.CertificateAuthorityARN != nil {
+			updateInput.SetCertificateAuthorityArn(*desired.ko.Spec.CertificateAuthorityARN)
+		}
```

Comment:
If it is nil, the API call will return an error since the CertificateAuthorityArn value wouldn't have been set. I followed the logic in the newCreateRequestPayload method